test cases/frameworks/25 hdf5/test.json

```diff
@@ -2,8 +2,7 @@
   "matrix": {
     "options": {
       "method": [
-        { "val": "pkg-config", "expect_skip_on_jobname": ["linux-gentoo-gcc"] },
-        { "val": "config-tool", "expect_skip_on_jobname": ["macos"] }
+        { "val": "pkg-config", "expect_skip_on_jobname": ["linux-gentoo-gcc"] }
```

Comment:
Note that this change is incorrect -- we define a matrix of jobs to run, and then mark each one as expected to skip on certain jobnames. You removed the matrix entry itself but instead, the skip value should be removed as a subkey.

See also #13920

Reply:
Since already merged as #13920, 0a5a992 was dropped.